community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NodeRelationshipCache.java

```diff
@@ -364,7 +364,7 @@ public int getCount( long nodeId, Direction direction )
         if ( isDense( array, nodeId ) )
         {   // Indirection into rel group cache
             long id = getRelationshipId( array, nodeId );
-            return id == EMPTY ? 0 : relGroupCache.getCount( id, direction );
+            return id == EMPTY ? 0 : relGroupCache.getAndResetCount( id, direction );
```

Comment:
And also fix the `NodeRelationshipCacheTests` that are failing because of this. Perhaps add one that explicitly tests for this.